bin/lean-gdb.py

```diff
@@ -1,3 +1,11 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2016 Microsoft Corporation. All rights reserved.
+# Released under Apache 2.0 license as described in the file LICENSE.
+#
+# Author: Sebastian Ullrich
```

Comment:
> gdb bin/lean should load the file automatically because of the file name (or print a message that the script must be whitelisted). I haven't tested it in CLion yet.

Ok, this only works if you run the lean binary in the bin directory, but not if you run the one in the cmake build directory.

> We actually had helper functions for that and I was confused why I couldn't find them until I noticed that you had removed them￼ .

Ooops, they were buried in a lot of commented-out code so I thought they were similarly unused.  I'll put them back.

> For expr, you'd usually want to use the Lean pretty printer,

Sometimes, yes.  But sometimes, I want to see the actual structure of an expression, and then you're completely lost, as the debugger will only show you the common fields for an expr since expr cells don't have RTTI info..

BTW, can you access the fields of the cell in a declaration from GDB?  I always get "incomplete type".